test_runner/regress/test_ingestion_layer_size.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder, wait_for_last_flush_lsn
+from fixtures.utils import human_bytes
+
+
+def test_ingesting_large_batches_of_images(neon_env_builder: NeonEnvBuilder, build_type: str):
+    """
+    Build a non-small GIN index which includes similarly batched up images in WAL stream as does pgvector
+    to show that we no longer create oversized layers.
+    """
+
+    if build_type == "debug":
+        pytest.skip("debug run is unnecessarily slow")
+
+    checkpoint_distance = 64 * 1024**2
+    env = neon_env_builder.init_start(
+        initial_tenant_conf={
+            "checkpoint_distance": f"{checkpoint_distance}",
+            "compaction_period": "0s",
+            "gc_period": "0s",
+            "compaction_threshold": "255",
+            "image_creation_threshold": "99999",
+        }
+    )
+
+    # build a 391MB gin index which exhibits the same behaviour as the pgvector with the two phase build
```

Comment:
```suggestion
    # build a larger than checkpoint_distance gin index
    # building a gin index exhibits the same behaviour as does pgvector with the two phase build
```

Reply:
With double the rows it was a 391MB but I must've forgotten to drop this. with 64MB checkpoint_distance it is enough to build half the size. Could we we could half it once more, but it's already 16s to build.

Reply:
[75496dd](https://github.com/neondatabase/neon/pull/7927/commits/75496dd7ddddd25dfe1fff07ac1e8a7291226ab0)